motoko/ios-notifications/README.md

```diff
@@ -0,0 +1,131 @@
+
+**Example of how this can be handled:**
+
+```ts
+async handleMultiPlatformLogin(): Promise<void> {
+    const key = await this.storage.get(KEY_STORAGE_KEY) ?? undefined;
+    const delegation = await this.storage.get(KEY_STORAGE_DELEGATION) ?? undefined;
+    const identityParam: IdentityParam = { key, delegation };
+    const preloadParam = Buffer.from(JSON.stringify(identityParam), "ascii").toString("base64");
+    const url = Auth.currentURL();
+
+    switch(this.loginType()) {
+        case AuthLoginType.Ios:
+            const iosCallback = new URL(url.searchParams.get(AuthLoginType.Ios) ?? "");
+            iosCallback.searchParams.append(Auth.identityPreloadProp, preloadParam);
+            // the redirect here triggers the app universal link
+            // such as dappexample://auth?_identity=...
+            // and this is what the app intercepts and handles
+            window.location.href = iosCallback.toString();
+            break;
+        default:
+            // desktop is enabled by default and doesn't need a special condition
+            break;
+    }
+}
+```
+
+## Notifications
+
+The ios app is prepared to receive notifications from remote APN servers. For the scope of this example we haven't setup our own notification server, instead, you can use the [send-notification.sh](send-notification.sh) script to trigger the notification with your own apple developer keys.
+
+These are the steps to show an IOS notification:
+
+1. When the app starts we use UNUserNotificationCenter to request the user for push notification permissions 
+1. With granted permissions a request to register for remote notifications is made
+1. A device id is made available with the remote call
+    1. for development purposes we print this value to the xcode console
+1. Execute the [send-notification.sh](send-notification.sh) script with the correct env variables and the notification will appear in your device
+    1. A physical ios device is required for this step since the simulator can't register remotely
+1. By clicking the notification the app will open in the about page
+
+## References
+
+For specific implementation details defer to:
+
+- [Readme: dapp-demo](dapp-demo/README.md)
+- [Readme: ios-dapp-demo](ios-dapp-demo/README.md) 
+
+## Disclaimer
+
+This is an example dapp that demonstrates the potential of integrating a dapp with native apps. 
+
+Please be mindful when considering this code for production and be mindful of:
+
+1. The integration with II in this example is using a universal links which is known to be safer than custom app schemes that are not bound to the appID. To enable this in your production code you should look at setting up [universal links](https://developer.apple.com/documentation/xcode/supporting-universal-links-in-your-app).
```

Comment:
Maybe a special section called `security considerations` would make sense? Given that _not_ using universal links is just insecure I would also word this prompt a bit stronger.

Reply:
Great suggestion, i've just added a `security considerations` section to the readme